apps/state/lib/state/shape.ex

```diff
@@ -222,16 +222,19 @@ defmodule State.Shape do
     |> Enum.sort_by(& &1.priority, &>=/2)
   end
 
-  defp find_override(id, overrides, suffix_overrides) do
-    if priority = Map.get(overrides, id) do
-      priority
-    else
-      Enum.find_value(suffix_overrides, fn {suffix, priority} ->
-        if String.ends_with?(id, suffix) do
-          priority
-        end
-      end)
-    end
+  defp find_override(id, prefix_overrides, suffix_overrides) do
+    Enum.find_value(
+      [{&String.starts_with?/2, prefix_overrides}, {&String.ends_with?/2, suffix_overrides}],
+      fn {check_fn, overrides} ->
+        find_override_priority(check_fn, overrides, id)
+      end
+    )
+  end
+
+  defp find_override_priority(overrides, check_fn, id) do
```

Comment:
You flipped the arguments. You call it as "(check_fn, overrides, id)", but it's defined as "(overrides, check_fn, id)".